sources/awskinesis/awskinesis.go

```diff
@@ -69,12 +82,56 @@ func (a AWSKinesis) validate() error {
 }
 
 // Fetch retrieves the next document from the awskinesis source
-func (a AWSKinesis) Fetch() ([]byte, error) {
-	// getrecord here
-	return nil, nil
+func (a AWSKinesis) Fetch(shardID uint) error {
+	// set up the shard iterator for our particular shardID
+	// NOTE: may want to make the ShardIteratorType into a config value
+	//       https://docs.aws.amazon.com/sdk-for-go/api/service/kinesis/#GetShardIteratorInput
+	iter, err := a.Service.GetShardIterator(
+		&kinesis.GetShardIteratorInput{
+			ShardId:           a.Shards[shardID].ShardId,
+			ShardIteratorType: aws.String("TRIM_HORIZON"),
+			StreamName:        aws.String(a.StreamName),
+		},
+	)
+	if err != nil {
+		return err
+	}
+
+	for {
```

Comment:
I think that this should be in the worker pool. The connector should be only about returning a message. Ideally would be to make this blocking call.

Reply:
This is the part that I’m confused about. We are consuming messages from a shard (each worker is assigned a dedicated shard). Techinically this function itself is blocking as it will loop pulling the next records from the queue (or wait for next set). I’m not convinced we give each individual record back to the job to process because that could end up being a lot of data...IMHO that defeats the purpose of setting up individual workers anyway.